ADALiOS/ADALiOS/ADAuthenticationRequest+AcquireToken.m

```diff
@@ -335,7 +354,7 @@ - (void) validatedAcquireTokenByRefreshToken:(NSString*)refreshToken
           additionalHeaders:nil
                  completion:^(NSDictionary *response)
      {
-         ADTokenCacheStoreItem* resultItem = (cacheItem) ? cacheItem : [ADTokenCacheStoreItem new];
+         ADTokenCacheStoreItem* resultItem = (cacheItem) ? [ADTokenCacheStoreItem initWithStoreItem:cacheItem] : [ADTokenCacheStoreItem new];
```

Comment:
The call stack is like:
(1)attemptToUseCacheItem()  -> (2)acquireTokenByRefreshToken() -> (3)validatedAcquireTokenByRefreshToken()

Variable *cacheItem* above is a parameter passed from (1) all the way to (3), however, such a parameter is changed in (3) because the above deleted line didn't make a copy of the parameter. In particular, when (3) set the resource property of *resultItem*, it changes the resource property of *cacheItem* as well.

Such a behaviour causes the following problem:
(1) is responsible for trying both RT and MRRT (if necessary), which is implemented in a recursive way. Function (1) assumes variable *cacheItem* should not be changed by anybody. It tells whether the cacheItem is a RT/MRRT by looking at the resource property being nil or not. If YES, it will stop the recursive call because MRRT should be the last token it try before asking user to log in again.

The above deleted line leads function (1) to misjudge whether cacheItem is RT or MRRT, therefore harm the recursive call terminating condition. In particular, it causes infinite retry for 500 error.

(PS: infinite recursive call happens in 500 error but not some other error code is because, 500 error will not cause any MRRT to be deleted, so function(1) can always find a MRRT for a retry. But the source of problem is still that function (3) changes the parameter *cacheItem*)